src/bionics.cpp

```diff
@@ -1535,14 +1497,18 @@ void bionic::deserialize( JsonIn &jsin )
     charge = jo.get_int( "charge" );
 }
 
-void player::introduce_into_anesthesia( time_duration const &duration )
+void player::introduce_into_anesthesia( time_duration const &duration,
+                                        bool external_kit )
 {
     add_msg_if_player( m_info,
                        _( "You type data into the console, configuring Autodoc to work with a CBM." ) );
     add_effect( effect_narcosis, duration );
     fall_asleep( duration );
     add_msg_if_player( m_info,
                        _( "Autodoc injected you with anesthesia, and while you were sleeping conducted a medical operation on you." ) );
+    if( external_kit ) {
```

Comment:
What is this?

Reply:
if you've already collected the anesthesia kit in another routine, there's no need to search for it here.
The logic of the change is right and necessary, the implementation is not great. I'll change that to pass the item_comp or item_loc or something like that.

Reply:
going back to the boolean, because although it's a little inelegant, it at least works. I cannot figure out how to pass an item_location to another function because I am not very good at c++ but I am very frustrated.